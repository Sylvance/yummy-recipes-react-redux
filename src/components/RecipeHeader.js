import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import RecipeFormInput from './RecipeFormInput'

export default class RecipeHeader extends Component {
  static propTypes = {
    addRecipe: PropTypes.func.isRequired,
    categoryid: PropTypes.number.isRequired,
    categoryname: PropTypes.string.isRequired
  }

  constructor(props) {
    super(props);
    this.state = {
      modal: false
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({
      modal: !this.state.modal
    });
  }

  handleSave = (categoryid, title, description) => {
    if (title.length !== 0) {
      this.props.addRecipe(categoryid, title, description)
      this.props.fetchRecipes(categoryid)
    }
    this.setState({
      modal: !this.state.modal
    });
  }

  render() {
    return (
      <header className="header">
        <h1>Recipes</h1>
        <h5>Category name : {this.props.categoryname}</h5>
        <Button color="primary" onClick={this.toggle}>Click here to add a recipe</Button>
        <Modal isOpen={this.state.modal} toggle={this.toggle} className={this.props.className}>
          <ModalHeader toggle={this.toggle}>Add Recipe</ModalHeader>
          <ModalBody>
            <RecipeFormInput onSave={this.handleSave} categoryid={this.props.categoryid}/>
          </ModalBody>
          <ModalFooter>
            <Button color="danger" onClick={this.toggle}>Cancel</Button>
          </ModalFooter>
        </Modal>
      </header>
    )
  }
}
