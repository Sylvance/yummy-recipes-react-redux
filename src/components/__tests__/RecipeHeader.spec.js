// Auto-generated do not edit

/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import RecipeHeader from '../RecipeHeader';

describe('RecipeHeader test', () => {
  it('RecipeHeader should match snapshot', () => {
    const component = renderer.create(
      <RecipeHeader
        addRecipe={() => {}}
        categoryid={42}
        categoryname={'defaultString'}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
