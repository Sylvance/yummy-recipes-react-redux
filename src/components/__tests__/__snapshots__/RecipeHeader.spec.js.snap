// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeHeader test RecipeHeader should match snapshot 1`] = `
<header
  className="header"
>
  <h1>
    Recipes
  </h1>
  <h5>
    Category name : 
    defaultString
  </h5>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    Click here to add a recipe
  </button>
</header>
`;
